### PR TITLE
Calculations measures

### PR DESCRIPTION
**Pull request by aljesusg**

### CODE REVIEW

File: spec/models/manageiq/providers/consumption/consumption_manager/showback_event_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+RSpec.describe ShowbackEvent, type: :model do
+  let(:vm)    { FactoryGirl.create(:vm_or_template) }
+  let(:event) { FactoryGirl.build(ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackEvent) }
```
Comment (sergio-ocon) [inline spec/models/manageiq/providers/consumption/consumption_manager/showback_event_spec.rb]: YOu can define the factory as :showback_measure_type, and use the class to define the actual class (better looking)

File: spec/factories/consumption_showback_measure_type.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackMeasureType, parent: :showback_measure_type, class: ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackMeasureType do
```
Comment (sergio-ocon) [inline spec/factories/consumption_showback_measure_type.rb]: Factory :consumtion_showback_meadure_type

File: app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
+    @accumulated.each do |key,dimensions|
+      dimensions.keys.each do |dim|
+        @accumulated[key][dim] = self.send("#{key}_#{dim}")
+      end
+    end
+    @accumulated
+  end
+
+  def dimension_acumulate
+    @accumulated = {}
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb]: @accumulated could be named stock
It doesn't sound right

File: app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb [hunk 4]
```diff
@@ -1,6 +1,37 @@
...
+  def event_object
+    type_obj.constantize.find(id_obj)
+  end
+
+  def metrics_time_range(start_time, end_time)
+    event_object.metrics.for_time_range(start_time, end_time)
+  end
+
+  def CPU_average
+    @metrics.average(:cpu_usage_rate_average)
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb]: total_time = old_time + new_time
average = (old_avg * old_time + new_avg * new_time)/ total_time

File: spec/models/manageiq/providers/consumption/consumption_manager/showback_event_spec.rb (hunk 1 above)
Comment (aljesusg) [inline spec/models/manageiq/providers/consumption/consumption_manager/showback_event_spec.rb]: If I define like this I can't use the methods of ManageIQ::Providers::Consumption::ConsumptionManager::ShowbackMeasureType

**Comment by aljesusg** [comment]
@sergio-ocon New release, comments, update data hash, new average calculation.

**Pull request merged**